pwiz_tools/Skyline/Model/DocNode.cs

```diff
@@ -1380,4 +1380,83 @@ public interface IPickedList
 
         bool AutoManageChildren { get; }
     }
+
+    public class DocNodePath
+    {
+        public PeptideGroupDocNode Protein { get; private set; }
+        public PeptideDocNode Peptide { get; private set; }
+        public TransitionGroupDocNode Precursor {get; private set; }
+        public TransitionDocNode Transition { get; private set; }
+        public SrmDocument Document { get; private set; }
+
+        private DocNodePath(PeptideGroupDocNode prot, PeptideDocNode pep, TransitionGroupDocNode pre, TransitionDocNode trans, SrmDocument doc)
+        {
+            Document = doc;
+            Protein = prot;
+            Peptide = pep;
+            Precursor = pre;
+            Transition = trans;
+        }
+
+        private static PeptideGroupDocNode GetProtein(SrmDocument doc, Peptide idPeptide, out PeptideDocNode peptide)
+        {
+            foreach (PeptideGroupDocNode prot in doc.Children)
+            {
+                peptide = prot.FindNode(idPeptide) as PeptideDocNode;
+                if (peptide != null)
+                {
+                    return prot;
+                }
+            }
+            peptide = null;
+            return null;
+        }
+
+        public static DocNodePath GetNodePath(Identity id, SrmDocument doc)
+        {
+            DocNodePath res = null;
+            switch (id)
+            {
+                case Transition idTransition:
+                {
+                    var prot = GetProtein(doc, idTransition.Group.Peptide, out var pep);
+                    if (prot != null && pep != null)

```

Comment:
I'd probably write this as:
```
if (prot == null || pep == null) 
{
   return null;
}
var pre = pep.FindNode(idTransition.Group) as TransitionGroupDocNode;
var trans = pre?.FindNode(idTransition) as TransitionDocNode;
return trans == null ? null : new DocNodePath(prot, pep, pre, trans, doc);
```
There's no need to set "res" and wait until the end of the function to return it. You might as well return the result as soon as you know what it is going to be.

Reply:
Fixed.